ir/src/tests/variables.rs

```diff
@@ -66,29 +162,121 @@ fn err_ic_variable_access_before_declaration() {
 }
 
 #[test]
-fn err_variable_vector_invalid_access() {
+fn err_variable_def_in_other_section() {
     let source = "
     const A = [[2, 3], [1, 0]]
     trace_columns:
         main: [clk]
     public_inputs:
         stack_inputs: [16]
     boundary_constraints:
+        let a = 1
         enf clk.first = 0
         enf clk.last = 1
     integrity_constraints:
+        enf clk' = clk + a";
+
+    let parsed = parse(source).expect("Parsing failed");
+    let result = AirIR::from_source(&parsed);
+    println!("{result:?}");
+    assert!(result.is_err());
+
+    let source = "
+    const A = [[2, 3], [1, 0]]
+    trace_columns:
+        main: [clk]
+    public_inputs:
+        stack_inputs: [16]
+    integrity_constraints:
+        let a = 1
+        enf clk' = clk + 1
+    boundary_constraints:
+        enf clk.first = 0
+        enf clk.last = a";
```

Comment:
Is it intentional to reverse the order of integrity and boundary constraints sections here?

Reply:
Oops, no! Good catch, thanks

Reply:
Oh wait, yes - it is. It needs to be reversed there in order to test defining the variable in the `integrity_constraints` and then using it in the `boundary_constraints` (out of scope)

Reply:
My mistake, I thought that it would work the other way around as well but you probably did it to make the example clearer but it wouldn't work the other way around.